src/modules/CSADigitizer/CSADigitizerModule.cpp

```diff
@@ -0,0 +1,337 @@
+/**
+ * @file
+ * @brief Implementation of charge sensitive amplifier digitization module
+ * @copyright Copyright (c) 2017-2020 CERN and the Allpix Squared authors.
+ * This software is distributed under the terms of the MIT License, copied verbatim in the file "LICENSE.md".
+ * In applying this license, CERN does not waive the privileges and immunities granted to it by virtue of its status as an
+ * Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+#include "CSADigitizerModule.hpp"
+
+#include "core/utils/unit.h"
+#include "tools/ROOT.h"
+
+#include <TFile.h>
+#include <TGraph.h>
+#include <TH1D.h>
+#include <TH2D.h>
+#include <TProfile.h>
+
+#include "objects/PixelHit.hpp"
+
+using namespace allpix;
+
+CSADigitizerModule::CSADigitizerModule(Configuration& config,
+                                               Messenger* messenger,
+                                               std::shared_ptr<Detector> detector)
+    : Module(config, std::move(detector)), messenger_(messenger), pixel_message_(nullptr) {
+    // Enable parallelization of this module if multithreading is enabled
+    enable_parallelization();
+
+    // Require PixelCharge message for single detector
+    messenger_->bindSingle(this, &CSADigitizerModule::pixel_message_, MsgFlags::REQUIRED);
+
+    // Seed the random generator with the global seed
+    random_generator_.seed(getRandomSeed());
+
+    // Set defaults for config variables
+    config_.setDefault<bool>("simple_parametrisation", true);
+    // defaults for the "advanced" parametrisation
+    config_.setDefault<double>("rise_time_constant", Units::get(1e-9, "s"));  
+    config_.setDefault<double>("feedback_time_constant", Units::get(10e-9, "s"));  // R_f * C_f
+    // for both cases
+    config_.setDefault<double>("feedback_capacitance", Units::get(5e-15, "C/V"));
+    // and for the advanced one
+    config_.setDefault<double>("krummenacher_current", Units::get(20e-9, "C/s"));
+    config_.setDefault<double>("detector_capacitance", Units::get(100e-15, "C/V"));
+    config_.setDefault<double>("amp_output_capacitance", Units::get(20e-15, "C/V"));
+    config_.setDefault<double>("transconductance", Units::get(50e-6, "C/s/V") );
+    config_.setDefault<double>("v_temperature", Units::get(25.7e-3, "eV"));  // Boltzmann kT at 298K
+
+    config_.setDefault<double>("amp_time_window", Units::get(0.5e-6, "s"));
+    config_.setDefault<double>("threshold", Units::get(10e-3, "V"));
+    config_.setDefault<double>("sigma_noise", Units::get(0.1e-3, "V"));
+    
+    config_.setDefault<bool>("output_pulsegraphs", false);
+    config_.setDefault<bool>("output_plots", config_.get<bool>("output_pulsegraphs"));
+    config_.setDefault<int>("output_plots_scale", Units::get(30, "ke"));
+    config_.setDefault<int>("output_plots_bins", 100);
+
+
+    // Copy some variables from configuration to avoid lookups:
+    tmax_ = config_.get<double>("amp_time_window");
```

Comment:
I suggest renaming this to `integration_time` which is both a common expression for this feature in front-ends and also a parameter name already ised in other parts of the framework for the same purpose (stopping the simulation eventually)